src/systemctl/systemctl.c

```diff
@@ -6298,6 +6298,43 @@ static int enable_unit(int argc, char *argv[], void *userdata) {
                 }
         }
 
+        if (!arg_quiet && !arg_no_block && streq(verb, "disable") && !install_client_side()) {
+                _cleanup_(bus_wait_for_jobs_freep) BusWaitForJobs *w = NULL;
+                const char* extra_args[4] = {};
+                int q, arg_count = 0;
+                sd_bus *bus;
+                char **name;
+
+                r = acquire_bus(BUS_MANAGER, &bus);
+                if (r < 0)
+                        return r;
+
+                r = bus_wait_for_jobs_new(bus, &w);
+                if (r < 0)
+                        return log_error_errno(r, "Could not watch jobs: %m");
+
+                if (arg_scope != UNIT_FILE_SYSTEM)
+                        extra_args[arg_count++] = "--user";
+
+                assert(IN_SET(arg_transport, BUS_TRANSPORT_LOCAL, BUS_TRANSPORT_REMOTE, BUS_TRANSPORT_MACHINE));
+                if (arg_transport == BUS_TRANSPORT_REMOTE) {
+                        extra_args[arg_count++] = "-H";
+                        extra_args[arg_count++] = arg_host;
+                } else if (arg_transport == BUS_TRANSPORT_MACHINE) {
+                        extra_args[arg_count++] = "-M";
+                        extra_args[arg_count++] = arg_host;
+                }
+
+                q = bus_wait_for_jobs(w, arg_quiet, extra_args);
```

Comment:
What is this trying to do? I don't see any `bus_wait_for_jobs_add()`. So this will not wait for anything... [bus_wait_for_jobs()](https://github.com/systemd/systemd/blob/1ab0a250c240d0255ac50224a7e799376878eb2b/src/shared/bus-unit-util.c#L1958) will return without doing anything, right?

I know you said the code might not be the best, but I was not expecting some lines that do nothing :-).